howtos/diffs/checkpointing.diff

```diff
@@ -11,23 +11,23 @@ index 5dc74db..975e50f 100644
  import jax
  from jax import random
 @@ -187,6 +188,9 @@ def train(train_ds, test_ds):
-   model = create_model(rng)
+   model = create_model(init_rng)
    optimizer = create_optimizer(model, FLAGS.learning_rate, FLAGS.momentum)
  
 +  # Restore saved model
 +  optimizer = checkpoints.restore_checkpoint(FLAGS.model_dir, optimizer)
 +
-   input_rng = onp.random.RandomState(0)
- 
    for epoch in range(1, num_epochs + 1):
-@@ -199,6 +203,10 @@ def train(train_ds, test_ds):
+     rng, input_rng = random.split(rng)
+     optimizer, train_metrics = train_epoch(
+@@ -198,6 +202,10 @@ def train(train_ds, test_ds):
      summary_writer.scalar('train_accuracy', train_metrics['accuracy'], epoch)
      summary_writer.scalar('eval_loss', loss, epoch)
      summary_writer.scalar('eval_accuracy', accuracy, epoch)
-+    
-+    # Save model at every epoch
-+    checkpoints.save_checkpoint(FLAGS.model_dir, optimizer, epoch, keep=3)
-+  
++
++  # Save model at every epoch
```

Comment:
Is this correct? The indentation look to have changed -- doesn't this now only save the model at the end of training?

Reply:
Yes, that sounds right; fixed.